### PR TITLE
add IFenwickTree interface

### PR DESCRIPTION
**Pull request by MikeHathaway**

### CODE REVIEW

File: src/IFenwickTree.sol [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * @dev   Used to efficiently track and retrieve lender and borrower positions in a pool.
+ * @dev   Nodes in the tree are equivalent to price buckets.
+ */
+interface IFenwickTree {
+
+    /***********************/
+    /*** State Variables ***/
+    /***********************/
+
+    // TODO: should scaling and values be internal to prevent external usage and avoid confusion on what they return?
```
Comment (EdNoepel) [inline src/IFenwickTree.sol]: Probably.

File: src/IFenwickTree.sol [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    // TODO: finish defining scaling factor
```
Comment (EdNoepel) [inline src/IFenwickTree.sol]: Feel free to crib off the sim's implementation, which is documented: https://github.com/ajna-finance/ajna-sim/blob/diez/ajna_sim/util.py#L123

**Review by EdNoepel — APPROVED** [review]
LGTM

File: src/IFenwickTree.sol (hunk 1 above)
Comment (MikeHathaway) [reply src/IFenwickTree.sol]: Updated accordingly

File: src/IFenwickTree.sol (hunk 2 above)
Comment (MikeHathaway) [reply src/IFenwickTree.sol]: Updated accordingly

**Pull request merged**